pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,163 @@
+		panic("Unexpected error!")
+	})
+
+	handler := HTTPPanicRecoveryHandler(panicHandlerFunc, logger.MustNewLogger("text", "info", "unix"))
+
+	req, err := http.NewRequest(http.MethodGet, "/", nil)
+	if err != nil {
+		require.NoError(t, err)
+	}
+
+	resp := httptest.NewRecorder()
+	require.NotPanics(t, func() {
+		handler.ServeHTTP(resp, req)
+	})
+
+	require.Equal(t, http.StatusInternalServerError, resp.Code)
+}
+
+func TestUnaryPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainUnaryInterceptor(
+			[]grpc.UnaryServerInterceptor{
+				grpc_recovery.UnaryServerInterceptor(
+					grpc_recovery.WithRecoveryHandlerContext(
+						PanicRecoveryHandler(logger.MustNewLogger("text", "info", "unix")),
+					),
+				),
+			}...,
+		),
+	}
+
+	srv := grpc.NewServer(serverOpts...)
+	t.Cleanup(srv.Stop)
+
+	openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+	go func() {
+		err := srv.Serve(listner)
+		if err != nil {
+			t.Fail()
+		}
+	}()
+
+	dialer := func(context.Context, string) (net.Conn, error) {
+		return listner.Dial()
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	opts := []grpc.DialOption{
+		grpc.WithContextDialer(dialer),
+		grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+	conn, err := grpc.DialContext(ctx, "", opts...)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		conn.Close()
+	})
+
+	cli := openfgav1.NewOpenFGAServiceClient(conn)
+
+	_, err = cli.Check(ctx, &openfgav1.CheckRequest{})
+	st, ok := status.FromError(err)
+	if !ok {
+		t.Fail()
+	}
+
+	require.Equal(t, http.StatusText(http.StatusInternalServerError), st.Message())
+}
+
+func TestStreamPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+	t.Cleanup(func() {
+		listner.Close()
+	})
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainStreamInterceptor(
+			[]grpc.StreamServerInterceptor{
+				grpc_recovery.StreamServerInterceptor(
+					grpc_recovery.WithRecoveryHandlerContext(
+						PanicRecoveryHandler(logger.MustNewLogger("text", "info", "unix")),
+					),
+				)}...,
+		),
+	}
+
+	srv := grpc.NewServer(serverOpts...)
+	t.Cleanup(srv.Stop)
+
+	openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+	go func() {
+		_ = srv.Serve(listner)
+	}()
+
+	dialer := func(context.Context, string) (net.Conn, error) {
+		return listner.Dial()
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	opts := []grpc.DialOption{
+		grpc.WithContextDialer(dialer),
+		grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+	conn, err := grpc.DialContext(ctx, "", opts...)
+	require.NoError(t, err)
+
+	cli := openfgav1.NewOpenFGAServiceClient(conn)
+	stream, err := cli.StreamedListObjects(ctx, &openfgav1.StreamedListObjectsRequest{})
+	require.NoError(t, err)
+
+	_, err = stream.Recv()
+	st, ok := status.FromError(err)
+	if !ok {
```

Comment:
```suggestion
	require.True(t, ok)
```
instead of `if !ok { t.Fail }`

Reply:
Done